lib/nerves_hub_link/device_channel.ex

```diff
@@ -106,7 +113,14 @@ defmodule NervesHubLink.DeviceChannel do
   def terminate(_reason, _state), do: NervesHubLink.Connection.disconnected()
 
   defp handle_join_reply(%{"firmware_url" => url} = update) when is_binary(url) do
-    UpdateManager.apply_update(update)
+    case NervesHubLinkCommon.Message.UpdateInfo.parse(update) do
+      %NervesHubLinkCommon.Message.UpdateInfo{} = info ->
```

Comment:
```suggestion
      {:ok, info} ->
```

In the other place, `parse` returns an `{:ok, info}`, so I assume this would here.

Reply:
oops!